test/parallel/xyk-pallet.API.fees.bothTokens.test.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ *
+ * @group xyk
+ * @group accuracy
+ * @group parallel
+ */
+import { getApi, getMangataInstance, initApi } from "../../utils/api";
+import { getCurrentNonce } from "../../utils/tx";
+import { ExtrinsicResult } from "../../utils/eventListeners";
+import { BN } from "@polkadot/util";
+import { Keyring } from "@polkadot/api";
+import { Assets } from "../../utils/Assets"
+import { AssetWallet, User } from "../../utils/User";;
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { SignerOptions } from "@polkadot/api/types";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import { RuntimeDispatchInfo } from "@polkadot/types/interfaces";
+import { MGA_ASSET_ID, KSM_ASSET_ID } from "../../utils/Constants";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.spyOn(console, "error").mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+process.env.NODE_ENV = "test";
+
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let firstCurrency: BN;
+let secondCurrency: BN;
+const first_asset_amount = new BN(50000);
+const second_asset_amount = new BN(50000);
+//creating pool
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+
+let cost: RuntimeDispatchInfo;
+
+const defaultCurrecyValue = new BN(250000);
+
+export async function UserMintCurrencyToken(
+  testUser: User,
+  sudo: User,
+  Currency: BN,
+  CurrencyValue: BN
+) {
+  await sudo.mint(
+    Currency,
+    testUser,
+    CurrencyValue
+  );
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  keyring = new Keyring({ type: "sr25519" });
+
+  sudo = new User(keyring, sudoUserName);
+
+  //add MGA tokens for creating pool.
+  await sudo.addMGATokens(sudo);
+
+  //add two curerncies and balance to sudo:
+  [firstCurrency, secondCurrency] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrecyValue, defaultCurrecyValue],
+    sudo
+  );
+
+  keyring.addPair(sudo.keyRingPair);
+
+  await (await getMangataInstance())
```

Comment:
Plz use this function. Just because it manages inside the sudo nonce for parallel testing. 

    await sudo.createPoolToAsset(
      amount1,
      amount2,
      tokenId1,
      tokenId2)
    );